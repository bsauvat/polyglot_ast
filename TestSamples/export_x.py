import polyglot

x = 42
polyglot.export_value(name="x", value=x)
polyglot.eval(path="impport_x.py", language="python")