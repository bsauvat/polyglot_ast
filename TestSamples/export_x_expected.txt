module
 import_statement
  import : import
  dotted_name
   identifier : polyglot
  expression_statement
   assignment
    identifier : x
    = : =
    integer : 42
   expression_statement
    polyglot_export_call
     attribute
      identifier : polyglot
      . : .
      identifier : export_value
      argument_list
       ( : (
       keyword_argument
        identifier : name
        = : =
        string
         string_start : "
         string_content : x
         string_end : "
        , : ,
        keyword_argument
         identifier : value
         = : =
         identifier : x
         ) : )
    expression_statement
     polyglot_eval_call
      module
       expression_statement
        assignment
         identifier : y
         = : =
         polyglot_import_call
          attribute
           identifier : polyglot
           . : .
           identifier : import_value
           argument_list
            ( : (
            keyword_argument
             identifier : name
             = : =
             string
              string_start : "
              string_content : x
              string_end : "
             ) : )
        expression_statement
         call
          print : print
          argument_list
           ( : (
           identifier : y
           ) : )
